### PR TITLE
Add program-prefix and dependence on flex to elfutils package

### PR DESCRIPTION
**Pull request by healther**

### CODE REVIEW

File: var/spack/repos/builtin/packages/elfutils/package.py [hunk 1]
```diff
@@ -40,10 +40,11 @@ class Elfutils(AutotoolsPackage):
     depends_on('autoconf', type='build')
 
     version('0.163',
-            git='git://git.fedorahosted.org/git/elfutils.git',
+            git='http://git.fedorahosted.org/git/elfutils.git',
             tag='elfutils-0.163')
 
     provides('elf')
+    depends_on('flex')  # needed in maintainer mode
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/elfutils/package.py]: This should be `type='build'`

**Comment by healther** [comment]
Do you want any additional changes?

File: var/spack/repos/builtin/packages/elfutils/package.py [hunk 2]
```diff
@@ -34,14 +34,14 @@ class Elfutils(AutotoolsPackage):
     version of elfutils."""
 
     homepage = "https://fedorahosted.org/elfutils/"
+    url = 'https://sourceware.org/elfutils/ftp/0.163/elfutils-0.163.tar.bz2'
+
+    version('0.163', '77ce87f259987d2e54e4d87b86cbee41')
 
     depends_on('libtool', type='build')
     depends_on('automake', type='build')
     depends_on('autoconf', type='build')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/elfutils/package.py]: @adamjstewart Other than `flex`, do we need any of these other dependencies now?

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/elfutils/package.py]: Nope. They were probably needed when we were checking out a tag straight from git, but the tarball we are switching to comes with a `configure` script, so these dependencies are no longer needed to generate one.

**Comment by citibeth** [comment]
@healther can you please see if it still works once you remove `libtool`,
`automake` and `autoconf` dependencies?

File: var/spack/repos/builtin/packages/elfutils/package.py (hunk 2 above)
Comment (davydden) [inline var/spack/repos/builtin/packages/elfutils/package.py]: @healther as per @adamjstewart 's comment, please remove those dependencies.

**Comment by healther** [comment]
I ran into a new problem here, this package does not build at all for me. It does not seem to be able to link against ```gettext```. I also looked this up on our cluster and it's not installed there -> I'm not sure whether this build at anytime. I'd be grate ful if someone tried the install on their own system, otherwise this should be ignored for now.

The error in ```spack-build.out``` is:
```
spack/env/gcc/gcc -std=gnu99 -Wall -Wshadow -Wformat=2 -Werror -Wunused -Wextra -Wstack-usage=262144   -fPIC   -o libelf.so -shared -o libelf.so -Wl,--whole-archive,libelf_pic.a,--no-whole-archive \
        -Wl,--version-script,/afs/kip.uni-heidelberg.de/user/wv385/Documents/git/spack/var/spack/stage/elfutils-0.163-72pp6vadezugf5nmy5gbqksrmpukksa3/elfutils-0.163/libelf/libelf.map,--no-undefined \
        -Wl,--soname,libelf.so.1,-z,defs,-z,relro 
libelf_pic.a(elf_error.os): In function `elf_errmsg':
elf_error.c:(.text+0x10f): undefined reference to `libintl_dgettext'
elf_error.c:(.text+0x13d): undefined reference to `libintl_dgettext'
elf_error.c:(.text+0x1c8): undefined reference to `libintl_dgettext'
collect2: error: ld returned 1 exit status
```

**Comment by adamjstewart** [comment]
Did you try adding a dependency on gettext? A lot of packages have missing dependencies because someone had it on their cluster when they wrote the package and didn't realize it was needed.

**Comment by healther** [comment]
I tried adding ```    depends_on('gettext')``` but that didn't change anything (which is admittedly weird). For now I decided to give up as I've exhausted my abilities to find the problem. 

Maybe I'll come back to it on the weekend.

**Pull request closed without merging**